WowPacketParserModule.V2_5_1_38707/Parsers/QueryHandler.cs

```diff
@@ -0,0 +1,290 @@
+                packet.ReadPackedGuid128("Player Guid");
+
+                packet.ReadUInt64("GuildClubMemberID");
+                packet.ReadUInt32("VirtualRealmAddress");
+
+                packet.ReadByteE<Race>("Race");
+                packet.ReadByteE<Gender>("Gender");
+                packet.ReadByteE<Class>("Class");
+                packet.ReadByte("Level");
+                packet.ReadByte("UnkBCC");
+
+                packet.ReadWoWString("Name", nameLen);
+            }
+        }
+
+        [HasSniffData]
+        [Parser(Opcode.SMSG_QUERY_CREATURE_RESPONSE)]
+        public static void HandleCreatureQueryResponse(Packet packet)
+        {
+            var entry = packet.ReadEntry("Entry");
+
+            CreatureTemplate creature = new CreatureTemplate
+            {
+                Entry = (uint)entry.Key
+            };
+
+            Bit hasData = packet.ReadBit();
+            if (!hasData)
+                return; // nothing to do
+
+            packet.ResetBitReader();
+            uint titleLen = packet.ReadBits(11);
+            uint titleAltLen = packet.ReadBits(11);
+            uint cursorNameLen = packet.ReadBits(6);
+            packet.ReadBit("Civilian");
+            creature.RacialLeader = packet.ReadBit("Leader");
+
+            var stringLens = new int[4][];
+            for (int i = 0; i < 4; i++)
+            {
+                stringLens[i] = new int[2];
+                stringLens[i][0] = (int)packet.ReadBits(11);
+                stringLens[i][1] = (int)packet.ReadBits(11);
+            }
+
+            for (var i = 0; i < 4; ++i)
+            {
+                if (stringLens[i][0] > 1)
+                {
+                    string name = packet.ReadDynamicString("Name", stringLens[i][0], i);
+                    if (i == 0)
+                        creature.Name = name;
+                }
+                if (stringLens[i][1] > 1)
+                {
+                    string nameAlt = packet.ReadDynamicString("NameAlt", stringLens[i][1], i);
+                    if (i == 0)
+                        creature.FemaleName = nameAlt;
+                }
+            }
+            creature.TypeFlags = packet.ReadUInt32E<CreatureTypeFlag>("Type Flags");
+            creature.TypeFlags2 = packet.ReadUInt32("Creature Type Flags 2");
+
+            creature.Type = packet.ReadInt32E<CreatureType>("CreatureType");
+            creature.Family = packet.ReadInt32E<CreatureFamily>("CreatureFamily");
+            creature.Rank = packet.ReadInt32E<CreatureRank>("Classification");
+            creature.PetSpellDataID = packet.ReadUInt32("PetSpellDataId");
+
+            creature.KillCredits = new uint?[2];
+            for (int i = 0; i < 2; ++i)
+                creature.KillCredits[i] = (uint)packet.ReadInt32("ProxyCreatureID", i);
+
+            var displayTotalCount = packet.ReadUInt32("DisplayIdCount");
+            packet.ReadSingle("TotalProbability");
+
+            for (uint i = 0; i < displayTotalCount; ++i)
```

Comment:
This looks broken to me if displayTotalCount > 4

Reply:
It is never > 4 in TBC.

Reply:
But I would like to preserve actual packet structure, not this kind of weird abomination whose purpose was maintaining compatibility with old database structure (those ids arent even saved in this handler!)

Reply:
I copied it straight from old handler iirc